crates/node-core/src/init.rs

```diff
@@ -96,16 +98,16 @@ pub fn init_genesis<DB: Database>(factory: ProviderFactory<DB>) -> Result<B256,
 }
 
 /// Inserts the genesis state into the database.
-pub fn insert_genesis_state<DB: Database>(
+pub fn insert_genesis_state<'a, 'b, DB: Database>(
     tx: &<DB as Database>::TXMut,
-    genesis: &reth_primitives::Genesis,
+    capacity: usize,
+    alloc: impl Iterator<Item = (&'a Address, &'b GenesisAccount)>,
```

Comment:
don't they have the same lifetime?

Reply:
It cannot be guaranteed. if they are both from the `Genesis` type then yes. but we cannot know that here.